includes/integrations/divi/scripts/builder-bundle.min.js

```diff
@@ -18,49 +18,101 @@ class ConvertKitForm extends React.Component {
+		const form = ( typeof this.props.form !== 'undefined' ? this.props.form : '' );
 
+		return ConvertKitDiviRenderModule(
+			window.ConvertkitDiviBuilderData.form,
+			form,
+			( form ? window.ConvertkitDiviBuilderData.form.fields.form.data.forms[ form ].name : '' ),
+			window.ConvertkitDiviBuilderData.form.fields.form.label
+		);
+
+	}
+
+}
+
+/**
+ * ConvertKit Product Divi Module.
+ */
+class ConvertKitDiviProduct extends React.Component {
+
+	/**
+	 * The Divi module name. Must match the slug defined in
+	 * the PHP class ConvertKit_Divi_Module_Product.
+	 * 
+	 * @since 	2.5.7
+	 */
+	static slug = 'convertkit_product';
+
+	/**
+	 * Renders the frontend output for this module.
+	 * 
+	 * @since 	2.5.7
+	 */
+	render() {
+
+		const product = ( typeof this.props.product !== 'undefined' ? this.props.product : '' );
+
+		return ConvertKitDiviRenderModule(
+			window.ConvertkitDiviBuilderData.product,
+			product,
+			( product ? window.ConvertkitDiviBuilderData.product.fields.product.values[ product ] : '' ),
+			window.ConvertkitDiviBuilderData.product.fields.product.label
+		);
+
+	}
+
+}
+
+/**
+ * Renders the Divi module for the given ConvertKit block (form, product, broadcast etc).
+ * and message, based on the Plugin and Divi module's configuration.
+ * 
+ * @since 	2.5.7
+ * 
+ * @param 	object 	block 			Block.
+ * @param 	string  value   		Selected value.
+ * @param 	string 	valueLabel 		Selected value's label.
+ * @param 	string 	resourceName 	Resource name (form,product).
+ */
+function ConvertKitDiviRenderModule( block, value, valueLabel, resourceName ) {
+
+	// If no Access Token has been defined in the Plugin, show a message in the module to tell the user what to do.
+	if ( ! block.has_access_token ) {
+		return React.createElement( 'div', {
+			className: 'convertkit-divi-module convertkit-no-content'
+		}, [
+			React.createElement( 'p', {}, block.no_access_token.notice ),
+			React.createElement( 'p', {}, block.no_access_token.instruction_text )
+		]  );
+	}
+
+	// If no resources exist in ConvertKit, show a message in the module to tell the user what to do.
+	if ( ! block.has_resources ) {
+		return React.createElement( 'div', {
+			className: 'convertkit-divi-module convertkit-no-content'
+		}, [
+			React.createElement( 'p', {}, block.no_resources.notice ),
+			React.createElement( 'p', {}, block.no_resources.instruction_text )
+		]  );
+	}
+
+	// Show instructions if no resource selected.
+	if ( value.length === 0 ) {
 		// Return module with text.
 		return React.createElement( 'div', {
 			className: 'convertkit-divi-module'
-		}, 'ConvertKit Form "' + form.name + '" selected. View on the frontend site to see the form.' );
-
+		}, 'Configure this module by clicking the settings cog.' );
 	}
 
+	// Return module with text.
+	return React.createElement( 'div', {
+		className: 'convertkit-divi-module'
+	}, 'ConvertKit ' + resourceName + ' "' + valueLabel + '" selected. View on the frontend site to see the product.' ); 
```

Comment:
Shouldn't product be generalized for both form and product?

Reply:
Yes - I spotted that yesterday working on the [broadcasts PR,](https://github.com/ConvertKit/convertkit-wordpress/pull/705/files#diff-c85ab5ae32c913ca5fd8a5d2628778a819dc6a5e2311fc1e1975ada4c063d149R132) which will correct this. Thanks for the heads up!